### PR TITLE
chore: cleanup auto-created namespaces during workspace deletion

### PR DESCRIPTION
**Pull request by amandavialva01**
## Ticket
DET-10225

## Description
Clean up auto-created namespaces made by determined during workspace deletion.

## Test Plan
CI passes.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dannysauer** [comment]
There might be a rebase in your future. 😂

**Comment by amandavialva01** [comment]
> There might be a rebase in your future. 😂

Gah, I had already rebased the upstream branch but just didn't push to remote 😂 I'll be more careful about making sure the branch I'm trying to merge into is up-to-date with the incoming branch. Order is restored 🙈

**Review by tara-det-ai — APPROVED** [review]
LGTM

File: master/internal/rm/multirm/multirm.go [hunk 1]
```diff
@@ -415,6 +415,16 @@ func (m *MultiRMRouter) VerifyNamespaceExists(namespaceName string,
...
 }
 
+func (m *MultiRMRouter) DeleteNamespace(namespaceName string) error {
+	err := m.fanOutRMCommand(func(rm rm.ResourceManager) error {
+		return rm.DeleteNamespace(namespaceName)
+	})
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (carolinaecalderon) [inline master/internal/rm/multirm/multirm.go:425]: little nit, just my style preference -- you could condense all this code and just return line 419 -- lines 422-425 don't add more information than the err value return on 419

**Review by carolinaecalderon — APPROVED** [review]
LGTM!

File: master/internal/rm/multirm/multirm.go (hunk 1 above)
Comment (amandavialva01) [reply master/internal/rm/multirm/multirm.go:425]: It's my style preference too 🙈😂, good catch!

**Pull request merged**